core/drivers/clk/clk.c

```diff
@@ -7,19 +7,43 @@
 #include <config.h>
 #include <drivers/clk.h>
 #include <kernel/boot.h>
+#include <kernel/mutex.h>
 #include <kernel/panic.h>
-#include <kernel/spinlock.h>
+#include <kernel/thread.h>
 #include <malloc.h>
 #include <stddef.h>
 #include <stdio.h>
 
 /* Global clock tree lock */
-static unsigned int clk_lock = SPINLOCK_UNLOCK;
+static struct mutex clk_lock = MUTEX_INITIALIZER;
 
 #ifdef CFG_DRIVERS_CLK_PRINT_TREE
 static SLIST_HEAD(, clk) clock_list = SLIST_HEAD_INITIALIZER(clock_list);
 #endif
 
+static void lock_clk_access(void)
```

Comment:
How about using the names `clk_lock()` and `clk_unlock()` instead? the `struct mutex clk_lock` above obviously needs to be renamed, perhaps to `clk_mutex`.